packages/broker/src/plugins/operator/fetchRedundancyFactor.ts

```diff
@@ -0,0 +1,24 @@
+import { Logger } from '@streamr/utils'
+import { Contract } from 'ethers'
+import { Operator, operatorABI } from '@streamr/network-contracts'
+import { OperatorServiceConfig } from './OperatorPlugin'
+
+const logger = new Logger(module)
+
+export async function fetchRedundancyFactor({
+    operatorContractAddress,
+    nodeWallet
+}: Pick<OperatorServiceConfig, 'operatorContractAddress' | 'nodeWallet'>): Promise<number> {
+    const operator = new Contract(operatorContractAddress, operatorABI, nodeWallet) as unknown as Operator
+    const metadataAsString = await operator.metadata()
+    let metadata: Record<string, unknown> = {}
+    if (metadataAsString.length > 0) {
+        try {
+            metadata = JSON.parse(metadataAsString)
+        } catch (e) {
+            logger.warn('Encountered malformed metadata', { operatorContractAddress, metadataAsString })
+        }
+    }
+    const redundancyFactor = Number(metadata.redundancyFactor)
+    return !isNaN(redundancyFactor) ? Math.max(redundancyFactor, 1) : 1
```

Comment:
Would it make sense that we'd throw if there is a invalid data? E.g. if a user configures redundancy factor to `50` meaning 50% redundancy, it would be good to throw show an error/warning?

Reply:
Hmm.. good question. It actually got me thinking and to write this test https://github.com/streamr-dev/network/pull/1820.

The redundancy factor here isn't a percentage but number of nodes to replicate to.